plugin/pkg/admission/exec/denyhostipc/admission.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package denyhostipc
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+func init() {
+	admission.RegisterPlugin("DenyExecOnHostIPC", func(client client.Interface, config io.Reader) (admission.Interface, error) {
```

Comment:
is there a reason we don't just add this to the `DenyExecOnPrivileged` admission controller? Not sure we need all that granularity, especially since we will likely have three types of elevated pods (privileged, hostIPC, hostPID) and two types of access (pods/exec and pods/attach). Not sure we want 6 admission controllers in play to protect that scenario